### PR TITLE
Tests for `/ipns` paths and `Cache-Control` HTTP header

### PR DESCRIPTION
**Pull request by hacdias**
This updates the conformance tests to bring them in line with the changes on the following PRs:

- https://github.com/ipfs/specs/pull/436
- https://github.com/ipfs/boxo/pull/459
- https://github.com/ipfs/kubo/pull/10115

Currently, we are not allowing `Cache-Control` headers in `/ipns` paths, but that's about to change.

### CODE REVIEW

**Comment by hacdias** [comment]
@lidel since the specifications say _SHOULD_ all the time regarding `Cache-Control` and `Last-Modified`, I find it hard to write meaningful tests. For `Cache-Control`, I changed from "must not have" to "either is empty, or is `public, max-age=*`". Regarding `Last-Modified`, I simply did not add checks. The spec says it should be the resolution time. So that means it can either be set or not. The only thing I could do is checking if the date is valid if present, but that seems very meta into HTTP.

**Review by lidel — APPROVED** [review]

**Comment by lidel** [comment]
Ack, I think this is sensible for now. We will refine once DNSLink  side of things is tackled in the future PRs.

**Comment by hacdias** [comment]
As this needs to be merged, and **published** before merging the Boxo and Kubo PRs, I am going ahead and merge this once CI completes (just removed the Boxo branch).

**Pull request merged**
